app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.kt

```diff
@@ -54,7 +55,8 @@ class AddEditTaskFragment : Fragment() {
         super.onActivityCreated(savedInstanceState)
         setupSnackbar()
         setupNavigation()
-        loadData()
+        this.setupRefreshLayout(viewDataBinding.refreshLayout)
+        viewModel.start(getTaskId())
```

Comment:
I still think start should be called in the initialization method of the viewmodel so that it isn't called again on a config change - is there a reason why it's being called from the creation methods?

Reply:
Yes, we need the ID which is not available during init in VM. It would be complicated to pass it in the constructor so this is the best way.

Reply:
Would it make sense to have some sort of check in start() where if the task ID hasn't changed, you don't reload?

Reply:
Already has this

```
        if (isDataLoaded) {
            // No need to populate, already have data.
            return
        }
```